source/hp/dof_handler.cc

```diff
@@ -1747,7 +1872,7 @@ namespace hp
             &(*tria)));
 
         std::vector<types::subdomain_id> saved_subdomain_ids;
-        if (shared_tria != nullptr && shared_tria->with_artificial_cells())
+        if (shared_tria->with_artificial_cells())
```

Comment:
You might want to put an `Assert(shared_tria != nullptr, ExcInternalError());` right after the `dynamic_cast` (which we know should succeed, but better check).